### DISCUSSION TITLE
PorousFlowPolyLineSink - specify injection temperature

### DISCUSSION THREAD

**Discussion by anupkgeos** [idaholab/moose#26272] [Q&A Modules: Porous Flow]
Hi all,
How can I inject fluid at a certain temperature using PorousFlowPolyLineSink? 
```
[./injection_T]
    type = PorousFlowPolyLineSink
    fluid_phase = 0
    variable = temperature
    SumQuantityUO = heat_enthalpy_in_inc
    line_base = '1 0.01 0.1 0.0'
    line_length = ${well_length}
    line_direction = '0 0 1'
    use_internal_energy = true
    p_or_t_vals = '-1e9 1e9'
    fluxes = '-${inj_ext_flux} -${inj_ext_flux}'
  [../]
```
Or is there any DiracKernel for a line source where injection temperature can be specified?

Thank you,
Anup

**Comment by lynnmunday** [comment]
These are documented on this page: 
https://mooseframework.inl.gov/modules/porous_flow/sinks.html

**Reply by anupkgeos** [reply]
with all line sinks, there is use option to `use_enthalpy` but no option to specify the injection temperature

**Comment by 1runer** [comment]
I don't think you can set a specific temperature for any kind of line object. Their value is always determined for a specific area/volume. Therefore a flux (e.g. energy) is needed to normalize the value over the area. 

I also doubt that this assumption is realistic. Let's start with a borehole. We know the temperature of the fluid at the top of the open borehole. This temperature then rises to the bottom of the borehole (or to reservoir temperature)depending on the rock properties. In modeling, a temperature at a point (open top hole) is often specified as Dirichlet-BC assuming a certain fluid flow. 

If you want to have a certain temperature a long a line, I would use the PorousFlowPolyLineSink for fluid flux only. For temperature I would define the line of interested as Nodeset (e.g. using BoundingBoxNodeSetGenerator) and add a Dirichlet-BC for temperature.
